lib/models/builder.js

```diff
@@ -46,13 +63,15 @@ class Builder extends CoreObject {
     this.environment = this.environment || 'development';
     process.env.EMBER_ENV = process.env.EMBER_ENV || this.environment;
 
-    let broccoli, options = {};
+    this.tree = this.readBuildFile(this.project.root);
+
+    let broccoli, options = {}, systemTemp = isExperimentEnabled('SYSTEM_TEMP');
     if (isExperimentEnabled('BROCCOLI_2')) {
       broccoli = require('broccoli');
       let tmpDir;
 
       // If not using system temp dir, compatability mode with broccoli-builder, tmp in root
-      if (!isExperimentEnabled('SYSTEM_TEMP')) {
+      if (!systemTemp) {
```

Comment:
Please do not cache experiment values. Each branch point should call `isExperimentEnabled` directly.